backend/src/main/java/storage/repository/GenericRepository.java

```diff
@@ -0,0 +1,240 @@
+        query.append( " WHERE id="+object.getId());
+
+        try {
+            PreparedStatement ps = getPreparedStatementFromJSON(jsonObject, query); //maps model data to the query
+            ps.executeUpdate();
+            ResultSet rs = ps.getGeneratedKeys();
+            rs.next();
+            int id = rs.getInt(1);
+            return this.getObject(id);
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    /***
+     * Deletes a specific model instance
+     * @param id
+     * @return
+     */
+    public T deleteObject(int id) {
+        String query = "DELETE * FROM "+ type.getSimpleName() +" WHERE id="+id;
+        System.out.println("Running query: "+query);
+        ResultSet rs =  this.dbInterface.executeQuerry(query);
+        try {
+            return this.getMatchJSON(rs).iterator().next();
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    /***
+     * Attempts to convert result set to a model instance
+     * @param rs the result set from a querry
+     * @return An Object instance of given generic type
+     * @throws SQLException if error with result set
+     */
+    private List<T> getMatchJSON(ResultSet rs) throws SQLException {
+
+        List<T> results = new ArrayList<T>();
+        T result = null;
+        while(rs.next()){
+            try {
+                result = type.newInstance();
+            } catch (InstantiationException e) {
+                e.printStackTrace();
+            } catch (IllegalAccessException e) {
+                e.printStackTrace();
+            }
+
+            JsonObject jsonModel = this.gson.toJsonTree(result).getAsJsonObject();
+
+            for (String key : jsonModel.keySet()){
+                JsonElement valWrapper = jsonModel.get(key);
+                JsonPrimitive val = null;
+
+                if(valWrapper.isJsonPrimitive()){
+                    val = valWrapper.getAsJsonPrimitive();
+                } else {
+                    throw new IllegalArgumentException("Invalid class");
+                }
+
+                if (val.isNumber()){
+                    jsonModel.addProperty(key, rs.getInt(key));
+                } else if (val.isString()){
+                    jsonModel.addProperty(key, rs.getString(key));
+                } else if (val.isBoolean()){
+                    jsonModel.addProperty(key, rs.getString(key));
+                }
+            }
+            results.add(gson.fromJson(jsonModel,this.type));
+        }
+        return results;
+    }
+
+    /***
+     * Maps a JSON objects field values to a prepared statement
+     * @param jsonObject
+     * @param query
+     * @return
+     * @throws SQLException
+     */
+    private PreparedStatement getPreparedStatementFromJSON(JsonObject jsonObject, StringBuilder query) throws SQLException {
+        System.out.println("Preparing query: "+query.toString());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)